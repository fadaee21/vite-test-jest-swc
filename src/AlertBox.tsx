import React from 'react';

type AlertBoxProps = {
  type: 'success' | 'error';
  message: string;
  closable?: boolean;
};

export const AlertBox: React.FC<AlertBoxProps> = ({ type, message, closable = false }) => {
  return (
    <div
      role="alert"
      style={{
        border: '1px solid',
        padding: '1rem',
        color: type === 'success' ? 'green' : 'red',
      }}
    >
      <strong>{type === 'success' ? 'Success:' : 'Error:'}</strong> {message}
      {closable && <button aria-label="close-button">x</button>}
    </div>
  );
};
