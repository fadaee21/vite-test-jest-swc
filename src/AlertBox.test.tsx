import { render, screen } from "@testing-library/react";
import { AlertBox } from "./AlertBox";

describe("AlertBox Component", () => {
  test("renders error message (RTL test)", () => { //React Testing Library (RTL)
    render(<AlertBox type="error" message="Something went wrong." closable />);

    // Regular assertion
    expect(screen.getByRole("alert")).toHaveTextContent(
      "Error: Something went wrong."
    );

    // Check for close button
    expect(screen.getByLabelText("close-button")).toBeInTheDocument();
  });

  // https://jestjs.io/docs/snapshot-testing
  test("matches snapshot: error + closable", () => {
    const { asFragment } = render(
      <AlertBox type="error" message="Error message" closable />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches snapshot: success + not closable", () => {
    const { asFragment } = render(
      <AlertBox type="success" message="It worked!" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
