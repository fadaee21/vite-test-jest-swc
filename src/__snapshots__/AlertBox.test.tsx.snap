// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox Component matches snapshot: error + closable 1`] = `
<DocumentFragment>
  <div
    role="alert"
    style="border: 1px solid; padding: 1rem; color: red;"
  >
    <strong>
      Error:
    </strong>
     Error message
    <button
      aria-label="close-button"
    >
      x
    </button>
  </div>
</DocumentFragment>
`;

exports[`AlertBox Component matches snapshot: success + not closable 1`] = `
<DocumentFragment>
  <div
    role="alert"
    style="border: 1px solid; padding: 1rem; color: green;"
  >
    <strong>
      Success:
    </strong>
     It worked!
  </div>
</DocumentFragment>
`;
